File: tournaments/lineEncoding/lineEncoding.js
function lineEncoding(s) {
    const n = s.length;
    const result = [];
    let count = 1;

    for (let i = 1; n > i; ++i) {
        if (s.charCodeAt(i - 1) - s.charCodeAt(i)) {
            if (1 < count) {
                result.push(count);
                count = 1;
            }
            result.push(s[i - 1]);
        } else {
            ++count
        }
    }

    if (1 < count) result.push(count);
    result.push(s[n - 1]);

    return result.join('');
}

File: tournaments/lineEncoding/lineEncoding.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub lineEncoding {
    my ($s) = @_;

    my @result;
    my @store = split('', $s);
    my $c = 1;

    for (my $i = 1; @store > $i; ++$i) {
        if (ord($store[$i - 1]) - ord($store[$i])) {
            if (1 < $c) {
                push(@result, $c);
                $c = 1;
            }
            push(@result, $store[$i - 1]);
        } else {
            ++$c;
        }
    }

    if (1 < $c) {
        push(@result, $c);
    }

    push(@result, $store[@store - 1]);

    return join('', @result);
}
